src/init/starship.bash

```diff
@@ -33,6 +33,9 @@ starship_preexec() {
 starship_precmd() {
     # Save the status, because commands in this pipeline will change $?
     STARSHIP_CMD_STATUS=$? STARSHIP_PIPE_STATUS=(${PIPESTATUS[@]})
+    if [[ ${BLE_ATTACHED-} ]]; then
```

Comment:
Maybe for consistency this would be more appropriate? 
```suggestion
    if [[ "${#BLE_PIPESTATUS[@]}" -gt "${#STARSHIP_PIPE_STATUS[@]}" ]]; then
```

Reply:
I'd say it doesn't work.

Inside ble.sh, `PIPESTATUS` contains a pipe status of random commands, so its content is unpredictable. With the current implementation of ble.sh, `PIPESTATUS` seems to be single-valued within the hook functions, but that can change in the future. Even if `PIPESTATUS` is ensured to be single-valued, the above condition doesn't work when the real pipe status of the previous user command is also single-valued (which is the majority of the cases).

Also, as far as the hook functions are called by ble.sh, `BLE_PIPESTATUS` is ensured to exist. Ah, but wait. An old version of ble.sh may not have the support. I've now checked the git blame, and `BLE_PIPESTATUS` was added two years ago. If we would care about it, I think the check could be

```suggestion
    if [[ ${BLE_ATTACHED-} && ${#BLE_PIPESTATUS[@]} -gt 0 ]]; then
```

Reply:
Added